pkg/cnb/builder_builder.go

```diff
@@ -70,7 +71,6 @@ func (bb *builderBlder) AddStack(baseImage v1.Image, clusterStack *buildapi.Clus
 	bb.os = file.OS
 	bb.baseImage = baseImage
 	bb.stackId = clusterStack.Status.Id
-	bb.runImage = clusterStack.Status.RunImage.Image
```

Comment:
Is `clusterStack.Status.RunImage.Image` fully qualified?

Reply:
Ya it's the resolved image

Reply:
It looks like it just the spec runImage which is not necessarily fully qualified. 

https://github.com/buildpacks-community/kpack/blob/03acfbb54d58af0a006da0c0d94a6d62e53c5735/pkg/cnb/remote_stack_reader.go#L71-L74

Reply:
I think that was intentional. The run image shouldn't necessarily be fully qualified. (E.g if you are using the builder with pack and the run image was updated)

Reply:
I suppose here, it needs to be relocated by kpack so, it is not necessarily useful.

Reply:
Which I guess exposes the downside with this approach. It is only really useful if the source of the stack is not available otherwise, it provides a worse experience.

Reply:
Ya I guess the trade off is full build reproducibility though which a floating tag doesn't guarantee